### ISSUE TITLE
gtk3 rubberband is invisible

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#17773]
### Bug report

**Bug summary**

All's in the title.
Note that https://github.com/matplotlib/matplotlib/pull/17772 is necessary to avoid throwing an exception due to a logic error, but even with it, the rubberband is invisible.

**Expected outcome**

**Matplotlib version**

  * Operating system: archlinux
  * Matplotlib version: master
  * Matplotlib backend (`print(matplotlib.get_backend())`): gtk3agg/gtk3cairo
  * Python version: 38
  * Jupyter version (if applicable):
  * Other libraries: pygobject 3.36.1, may be worth trying earlier ones... (if it's a regression there this may not be release critical)

**Comment by QuLogic** [comment]
With #17772, the rubberband works for me, but it is not perfect, as it flashes and disappears if you stop moving the cursor.

**Comment by anntzer** [comment]
A bisect suggest this has been broken ever since https://github.com/matplotlib/matplotlib/pull/10209(!) and that the correct fix is along the lines of restoring
```patch
diff --git i/lib/matplotlib/backends/backend_gtk3.py w/lib/matplotlib/backends/backend_gtk3.py
index 31d20f3e6..62d85fe39 100644
--- i/lib/matplotlib/backends/backend_gtk3.py
+++ w/lib/matplotlib/backends/backend_gtk3.py
@@ -294,6 +294,9 @@ class FigureCanvasGTK3(Gtk.DrawingArea, FigureCanvasBase):
         # docstring inherited
         if self.is_drawable():
             self.queue_draw()
+            # do a synchronous draw (its less efficient than an async draw,
+            # but is required if/when animation is used)
+            self.get_property("window").process_updates(False)
 
     def draw_idle(self):
         # docstring inherited
```
but I guess it seems costly to always do this, may be nicer to only do this on draw_rubberband but this doesn't seem to work there...

**Comment by tacaswell** [comment]
I merged #17772 to bank the crash fix.
